Sources/Build/BuildPlan.swift

```diff
@@ -123,7 +123,27 @@ extension BuildParameters {
     public func targetTripleArgs(for target: ResolvedTarget) throws -> [String] {
         var args = ["-target"]
         // Compute the triple string for Darwin platform using the platform version.
-        if triple.isDarwin() {
+        if triple.isIOS() {
+            guard let iOSSupportedPlatform = target.platforms.getDerived(for: .iOS) else {
+                throw StringError("the target \(target) doesn't support building for iOS")
+            }
+            args += [triple.tripleString(forPlatformVersion: iOSSupportedPlatform.version.versionString)]
+        } else if triple.isWatchOS() {
+            guard let watchOSSupportedPlatform = target.platforms.getDerived(for: .watchOS) else {
+                throw StringError("the target \(target) doesn't support building for watchOS")
+            }
+            args += [triple.tripleString(forPlatformVersion: watchOSSupportedPlatform.version.versionString)]
+        } else if triple.isTVOS() {
+            guard let tvOSSupportedPlatform = target.platforms.getDerived(for: .tvOS) else {
+                throw StringError("the target \(target) doesn't support building for tvOS")
+            }
+            args += [triple.tripleString(forPlatformVersion: tvOSSupportedPlatform.version.versionString)]
+        } else if triple.isDriverKit() {
+            guard let driverKitSupportedPlatform = target.platforms.getDerived(for: .driverKit) else {
+                throw StringError("the target \(target) doesn't support building for DriverKit")
+            }
+            args += [triple.tripleString(forPlatformVersion: driverKitSupportedPlatform.version.versionString)]
+        } else if triple.isDarwin() {
```

Comment:
Could we factor this a little differently, there's a lot of redundancy here. I think we could have an accessor for the platform and a display string on the triple and then do something like this (pseudo-code):

```
if let supportedPlatform = target.platforms.getDerived(for: triple.platform) {
  args += [triple.tripleString(forPlatformVersion: supportedPlatform.version.versionString)]
} else {
  throw StringError("the target \(target) doesn't support building for \(triple.platformDisplayString)")
}
```

Reply:
I'm pretty new to Swift so I had no luck with it

Reply:
Tried again with a little bit more time in hand